### DISCUSSION TITLE
Unit 5 - Exercise 1

### DISCUSSION THREAD

**Discussion by DSJohannes** [Lightning-AI/dl-fundamentals#40] [Exercises] [answered]
Dear Mr. Raschka,

I changed the classifier to a regression model and got the warning that the shape of the logits differ from that of the true_labels so the mse_loss results could be wrong. 
In the shared_step function I changed the order of the loss and predicted_labels so that the predicted_labels variable comes first and I changed the definition of the variable to: 
predicted_labels = torch.flatten(logits)

after that I changed the loss to:
loss = F.mse_loss(predicted_labels, true_labels)

Now I don't get the Warning and have reasonable results. I only wanted to ask if that solution is right or did I make a mistake?

Thanks for answering my question

Johannes

**Comment by rasbt** [comment]
Hi there, on the one hand, there is nothing wrong with what you do, but on the other hand, I wonder why you need the `flatten` function on the logits. How would your output layer look like?

**Reply by DSJohannes** [reply, marked as answer]
Hello,

thanks for your reply.
That's the error message:
UserWarning: Using a target size (torch.Size([32])) that is different to the input size (torch.Size([32, 1])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
loss = F.mse_loss(logits, true_labels)

This suggests that the logits have two dimension as the true_labels only have one dimension. This is confirmed when I look at the Tensors so I flattend the one with two dimensions and solved the error message. I only wondered if that leads to errors which I haven't seen yet.

My Outputlayer receives 5 features and gives 1 value out as I'm doing a regression task.

**Reply by rasbt** [reply]
You are absolutely correct with the flattening to bring it from `torch.Size([32, 1]` to `torch.Size([32]`. Just wanted to make sure your output layer sizes looked correct ^^

**Reply by DSJohannes** [reply]
Ok, thanks for your help.
